commit 35ec6c7689fabf79260e5966fd6a3d57811a8611 (refs/changes/27/627/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-07T16:45:29-07:00 (3 years ago)
    
    DOCKER-946 docker pull authentication errors are no longer reported
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/lib/backends/sdc/images.js b/lib/backends/sdc/images.js
index b266573..54ae62d 100644
--- a/lib/backends/sdc/images.js
+++ b/lib/backends/sdc/images.js
@@ -1415,13 +1415,14 @@ function pullImage(opts, callback) {
                 status: format('Error pulling image: (%s) %s host not found',
                     err.code, opts.rat.index.name)
             };
-        } else if (err.code === 'UnauthorizedError') {
+        } else if (err.code === 'UnauthorizedError'
+                || err.code === 'Unauthorized') {
             payload = {
                 status: format('Unauthorized error from registry %s trying to '
                     + 'pull %s (%s)', opts.rat.index.name, opts.rat.remoteName,
                     opts.req_id)
             };
-        } else if (err.code === 'NotFoundError') {
+        } else if (err.code === 'NotFoundError' || err.code === 'NotFound') {
             // E.g. `docker pull quay.io/no-such-user`.
             payload = {
                 status: format('Not Found (404) error from registry %s trying '
